src/solve.jl

```diff
@@ -68,15 +75,22 @@ function _get_alg_nn(args...; kwargs...)
     if haskey(kwargs, :alg_nn)
         solver = kwargs[:alg_nn]
     else
-        throw(ArgumentError("the solver for the controller `alg_nn` should be specified, but was not found"))
+        throw(ArgumentError("the solver for the controller `alg_nn` should be "*
+                            "specified, but was not found"))
     end
     return solver
 end
 
+# flowpipe with corresponding iteration
+struct ControlFlowpipe{FT<:Flowpipe}
```

Comment:
we could do this, but then loose some nice features of flowpipes (algebra, iteration, plotting and such). i would suggest to store the metadata in the extension dictionary of an existing flowpipe struct, they always have such field https://github.com/JuliaReach/ReachabilityAnalysis.jl/blob/master/src/Flowpipes/Flowpipe.jl#L24

if it helps to query the iteration, we can have a helper function `iteration(F::Flowpipe) = get(F.ext, F.ext[:iteration], nothing)`

Reply:
This type is only an internal storage for the waiting list. The result is a normal `MixedFlowpipe`. I think it is safer to have an internal wrapper here instead of modifying the flowpipes unless this information is relevant.

Reply:
I renamed it to `WaitingListElement` to avoid confusion.